bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -204,14 +208,38 @@ def execute_graph(self) -> None:
                          computation_graph_callable=computation_graph_callable,
                          initialization_fn=self.backend.initialization)
 
-        # Execute graph distributedly and return the aggregated results from all
-        # tasks
-        returned_values = self.backend.ProcessAndMerge(self._build_ranges(), mapper, distrdf_reducer)
+        # List of action nodes in the same order as values
+        local_nodes = self._get_action_nodes()
+
+        # Prepare a dictionary with additional information for live visualization
+        drawables_dict_update = {
+            # Key: node_id
+            node.node_id: (
+                # Tuple containing:
+                # 1. Callback functions passed by the user
+                self.drawables_dict[node.node_id],
+                # 2. Index of the node in the local_nodes list
+                i,
+                # 3. Name of the operation associated with the node
+                node.operation.name
+            )
+            for i, node in enumerate(local_nodes)
+            # Filter: Only include nodes requested by the user
+            if node.node_id in self.drawables_dict
+        }
```

Comment:
This logic should be guarded by some `if` statement. Looking at the `LiveVisualize` implementation in `__init__.py` I would say
```py
if self.drawables_dict is not None: # this is how we know the user requested LiveVisualize
```